js/src/sheet.ts

```diff
@@ -273,16 +280,55 @@ let SheetView = widgets.DOMWidgetView.extend({
             break;
         }
     },
-    _build_table() {
-        return Promise.resolve(new Handsontable(this.el, extend({
+    async _build_widgets_views() {
+        let data = this.model.data;
+        let rows = data.length;
+        let cols = data[0].length;
+        let widget_view_promises = {}
+        for (let row = 0; row < rows; row++) {
+            for (let col = 0; col < cols; col++) {
+                if (data[row][col] && data[row][col].options['type'] == 'widget') {
+                    let widget = data[row][col].value;
+                    let previous_view = this.widget_views[[row, col]];
+                    if (previous_view) {
+                        if(previous_view.model.cid == widget.cid) { // if this a proper comparison?
+                            widget_view_promises[[row, col]] = Promise.resolve(previous_view)
+                        } else {
+                            previous_view.remove()
+                            previous_view = null;
+                        }
+                    }
+                    if (!previous_view && widget && widget.widget_manager) {
+                        widget_view_promises[[row, col]] = widget.widget_manager.create_view(widget)
+                    }
+                }
+            }
+        }
+        for (let key in this.widget_views) {
+            if(this.widget_views.hasOwnProperty(key)) {
+                let [row, col] = key;
+                let widget_view = this.widget_views[key];
+                if(data[row][col] && data[row][col].value && data[row][col].value.cid == widget_view.model.cid) {
+                    // good, the previous widget_view should be reused
+                } else {
+                    // we have a leftover view from the previous run
+                    widget_view.remove()
```

Comment:
It's this code which is triggered too much. The view is removed at every update of the slider. (tested with a `console.log`)